lib/smartcar.rb

```diff
@@ -39,45 +25,154 @@ class AuthenticationError < ExternalServiceError; end
   class BadRequestError < ExternalServiceError; end
```

Comment:
this error class and the ones above (minus `ConfigNotFound` are not used anywhere